File: longest_substring_without_repeating_characters.cpp
// https://leetcode.com/problems/longest-substring-without-repeating-characters/
#include <vector>
#include <algorithm>

class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        if (s.length() == 0) {
            return 0;
        }
        vector<int> v{1};
        int i = 0;
        int j = 1;
        while(j < s.length()){
            // cout<<s.substr(i,j-i)<<" j= "<<j<<"i= "<<i<<endl;
            if ( s.substr(i,j-i).find(s[j]) != std::string::npos ){
                // cout<<"Inside If: "<<s.substr(i,j-i)<<" j= "<<j<<"i= "<<i<<endl;
                v.push_back(s.substr(i,j-i).length());
                // for(auto k:v){
                //     cout<<k<<" ";
                // }
                // cout<<endl;
                while(s[i] != s[j]){
                    i++;
                }
                i++;
                // cout<<"j = "<<j<<endl;
                // cout<<"i = "<<i<<endl;
            }
            // cout<<"\n\n";
            j++;
        }
        v.push_back(s.substr(i,j-i).length());
        return *std::max_element(begin(v),end(v));
    }
};

File: longest_substring_without_repeating_characters.py
# https://leetcode.com/problems/longest-substring-without-repeating-characters/ 
class Solution(object):
    def lengthOfLongestSubstring(self, st):
        """
        :type s: str
        :rtype: int
        """
        i = 0
        j = 1
        if st == '': return 0;
        
        le = [1]
        while j < len(st):
            if st[j] in st[i:j]:
                le.append(len(st[i:j]))
                while(st[i]!=st[j]):
                    i += 1
                i+=1
            j += 1
            
        le.append(len(st[i:j]))
                
        return max(le)